### ISSUE TITLE
Fails to compile with `zstd` feature enabled

### ISSUE THREAD

**Issue by davidrusu** [spacejam/sled#329]
Noticed this while working on #328. This will be fixed once that PR is merged but I wanted to make a separate issue to track this problem so that things don't slip by.

1. sled version: 0.15.17
2. rustc version: rustc 1.28.0-nightly (01cc982e9 2018-06-24)
3. operating system: macOS
4. logs: 
```
   Compiling pagecache v0.4.5
error[E0425]: cannot find value `len` in this scope
   --> /Users/davidrusu/.cargo/registry/src/github.com-1ecc6299db9ec823/pagecache-0.4.5/src/io/reader.rs:135:21
    |
135 |                     len,
    |                     ^^^ not found in this scope

error[E0425]: cannot find value `len` in this scope
   --> /Users/davidrusu/.cargo/registry/src/github.com-1ecc6299db9ec823/pagecache-0.4.5/src/io/reader.rs:138:52
    |
138 |                 Ok(LogRead::Flush(header.lsn, buf, len))
    |                                                    ^^^ not found in this scope

error[E0599]: no method named `get_zstd_compression_factor` found for type `config::Config` in the current scope
   --> /Users/davidrusu/.cargo/registry/src/github.com-1ecc6299db9ec823/pagecache-0.4.5/src/io/iobuf.rs:225:45
    |
225 |             compress(&*raw_buf, self.config.get_zstd_compression_factor())
    |                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /Users/davidrusu/.cargo/registry/src/github.com-1ecc6299db9ec823/pagecache-0.4.5/src/config.rs:233:1
    |
233 | pub struct Config {
    | ----------------- method `get_zstd_compression_factor` not found for this
    |
    = help: did you mean `set_zstd_compression_factor`?

error[E0277]: the size for value values of type `[u8]` cannot be known at compilation time
   --> /Users/davidrusu/.cargo/registry/src/github.com-1ecc6299db9ec823/pagecache-0.4.5/src/io/iobuf.rs:223:13
    |
223 |         let buf = if self.config.use_compression {
    |             ^^^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `[u8]`
    = note: to learn more, visit <https://doc.rust-lang.org/book/second-edition/ch19-04-advanced-types.html#dynamically-sized-types--sized>
    = note: all local variables must have a statically known size

error[E0614]: type `[u8]` cannot be dereferenced
   --> /Users/davidrusu/.cargo/registry/src/github.com-1ecc6299db9ec823/pagecache-0.4.5/src/io/iobuf.rs:247:48
    |
247 |         out[MSG_HEADER_LEN..].copy_from_slice(&*buf);
    |                                                ^^^^

error: aborting due to 5 previous errors
```
